DOCS-12525 [EDITS]Copy to edit (#180)
mulesoft/docs-mule-sdk: modules/ROOT/pages/rest-sdk/edits/project-setup.adoc

= Setup and Installation
ifndef::env-site,env-github[]
include::../_attributes.adoc[]
endif::[]

> To get started with REST SDK, ensure you meet the prerequisites and then follow the

Reviewer: GLOBAL per style ref., "ensure" > "ensure that" (same with "verify")
```suggestion
To get started with REST SDK, ensure that your environment meets the installation prerequisites and then follow the
```

> steps in this section to install the REST SDK plugin.

Reviewer: ```suggestion
steps to install the REST SDK plugin.
```
Per style ref., we try to avoid self-referential locators like "in this section" or "in the following topic" or similar so that the content is reusable.

== Prerequisites

> Before you start using REST SDK, verify that you have the tools needed:

Reviewer: ```suggestion
Before you install and use REST SDK, verify that your environment includes the following required tools:
```

* Java 8 SDK

> * Apache Maven (3.3.9 or newer recommended)

Reviewer: ```suggestion
* Apache Maven (3.3.9 or later)
```

Reviewer: See style ref. content about recommendations.

== Install the REST SDK Plugin

> Install the https://repository-master.mulesoft.org/nexus/service/local/repositories/releases/content/com/mulesoft/connectors/rest-connect-cli/1.0.0-RC2/rest-connect-cli-1.0.0-RC2-jar-with-dependencies.jar[REST SDK plugin] to use REST SDK. The plugin is a distribution

Reviewer: Because this is a single-step task topic, I suggest isolating the user action from the conceptual information:
```suggestion
Install the https://repository-master.mulesoft.org/nexus/service/local/repositories/releases/content/com/mulesoft/connectors/rest-connect-cli/1.0.0-RC2/rest-connect-cli-1.0.0-RC2-jar-with-dependencies.jar[REST SDK plugin] to use REST SDK. 
+
The plugin is a distribution
```

> of one or more related mojos (Maven plain Old Java Objects). Each mojo is an

Reviewer: ```suggestion
of one or more related _mojos_ (Maven plain Old Java Objects). Each mojo is an
```
HOWEVER...I looked this up, and I see that it is often all lowercase...but it makes more sense to me grammatically to present this as MOJO (as some sources do), because it is an initialism for Maven Old Java Objects. Let me know what you decide and I will add it to our Term Presentation List.

Author: Interesting as I believe it was actually _mojos_ but I changed it to mojos. But it looks like since it is an initialism, we should capitalize it as MOJO. Thank you for adding it to the Term Presentation List.

executable goal in Maven.

> If you have access to the MuleSoft's repositories, Maven automatically downloads

Reviewer: ```suggestion
If you have access to the MuleSoft repositories, Maven automatically downloads
```

the plugin when you try to use it.

> === Get Access to MuleSoft's Maven Repository

Reviewer: ```suggestion
=== Get Access to the MuleSoft Maven Repository
```

Reviewer: Per style ref., do not make trademarked names possessive.

> MuleSoft repositories are public, however, you must to configure Maven to access

Reviewer: ```suggestion
Although MuleSoft repositories are public, you must configure Maven to access
```

> the repositories.

Reviewer: ```suggestion
them.
```

> . Access the `settings.xml` Maven configuration file. +
> Linux: `Linux: ~/.m2/settings.xml` +
> Mac: `Mac: ~/.m2/settings.xml` +
> Windows: `C:\Users\<UserName>\.m2\settings.xml`

Reviewer: ```suggestion
. Access the appropriate `settings.xml` Maven configuration file: +
* Linux: `Linux: ~/.m2/settings.xml` +
* Mac: `Mac: ~/.m2/settings.xml` +
* Windows: `C:\Users\<UserName>\.m2\settings.xml`
```

. In `settings.xml`, add a new plugin repository to the `<profile>` section, for example:

+
[source,xml]
----
    <profiles>
        <profile>
            <id>default-profile</id>
            <activation>
                <activeByDefault>true</activeByDefault>
            </activation>

            <pluginRepositories>
                <pluginRepository>
                    <id>mulesoft-releases</id>
                    <name>Mule Releases Repository</name>
                    <url>https://repository.mulesoft.org/nexus/content/repositories/releases/</url>
                </pluginRepository>
            </pluginRepositories>

        </profile>
    </profiles>
----

== What's Next

> Now that you've set up your project, proceed to

Reviewer: Strictly my own preference, to avoid "now" and "proceed to"...
```suggestion
After you've set up your project, you can xref:rest-sdk/scaffold-connector-descriptor.adoc[scaffold a connector descriptor].
```
